source/extensions/filters/http/proto_api_scrubber/scrubbing_util_lib/field_checker.cc

```diff
@@ -10,10 +12,63 @@ namespace HttpFilters {
 namespace ProtoApiScrubber {
 
 FieldCheckResults FieldChecker::CheckField(const std::vector<std::string>&,
-                                           const Protobuf::Field*) const {
+                                           const Protobuf::Field* field) const {
+  // Currently, this method only checks the top level request/response fields and hence, the field
+  // name itself represents the field_mask.
+  std::string field_mask = field->name();
+
+  MatchTreeHttpMatchingDataSharedPtr match_tree =
+      (scrubber_context_ == ScrubberContext::kRequestScrubbing)
```

Comment:
I'm guessing `ScrubberContext` is an enum. I suggest using a switch statement here and explicitly handle all the cases.